src/compiler/parser.ts

```diff
@@ -6659,27 +6736,16 @@ namespace ts {
                     if (!tagName) {
                         return false;
                     }
-
                     switch (tagName.text) {
                         case "type":
-                            if (parentTag.jsDocTypeTag) {
-                                // already has a @type tag, terminate the parent tag now.
-                                return false;
-                            }
-                            parentTag.jsDocTypeTag = parseTypeTag(atToken, tagName);
-                            return true;
+                            return !alreadyHasTypeTag && target === PropertyLikeParse.Property && parseTypeTag(atToken, tagName);
                         case "prop":
                         case "property":
-                            const propertyTag = parseParameterOrPropertyTag(atToken, tagName, /*shouldParseParamTag*/ false) as JSDocPropertyTag;
-                            if (propertyTag) {
-                                if (!parentTag.jsDocPropertyTags) {
-                                    parentTag.jsDocPropertyTags = <MutableNodeArray<JSDocPropertyTag>>[];
-                                }
-                                (parentTag.jsDocPropertyTags as MutableNodeArray<JSDocPropertyTag>).push(propertyTag);
-                                return true;
-                            }
-                            // Error parsing property tag
-                            return false;
+                            return target === PropertyLikeParse.Property && parseParameterOrPropertyTag(atToken, tagName, target);
+                        case "arg":

```

Comment:
Yes, `JSDocTag.tagName: Identifier`, so it looks like it's preserved on all JSDocTags.

A more useful lint rule would be to look for mismatched `typedef/param and param/property` nestings, since the compiler currently ignores them:

```js
/**
 * @typedef {object} x
 * @param {string} x.a - WRONG! (but ignored by the compiler)
 *
 * @param {object} y
 * @property {string} y.a - WRONG! (but ignored by the compiler)
 */
```